src/main/io/osd_elements_impl.h

```diff
@@ -0,0 +1,918 @@
+/*
```

Comment:
this should be .c file, exporting only osdElementDrawFunction

Reply:
Agreed there, this is breaking with the file structuring convention, and will throw people off because of this.

Reply:
I followed the model set by `sensors/gyro_filter_impl.h` and how it was included in `sensors/gyro.c`.

I don't mind changing but not sure exactly what you want. Just change the name to `osd_elements.c` and include it as is?

Or convert to `osd_elements.c` and `osd_elements.h` and include that way? This really won't work as all the display functions depend on static variables and other utility functions in `osd.c`.

Reply:
`sensors/gyro_filter_impl.h` not a good example to follow: First of all it is not a header file (which by definition should not contain any statements), and should be `gyro_filter_impl.c` by all means. Second, using code snippets to construct compilation units is not a good pattern - compilation units should be independent. `sensors/gyro_filter_impl.h` was done the way it is done to get around the fact that we had 2 more or less identical copies of the same complex sequence of code (replicated purely for optimisation reasons), and got broken a number of times when one copy was changed and the other one forgotten. If it wasn't for that we would never have gone for this structure.

> Or convert to osd_elements.c and osd_elements.h and include that way? This really won't work as all the display functions depend on static variables and other utility functions in osd.c.

In that case the static variables and utility functions should be moved into `osd_elements.c`, to make it a compilation unit. Or turn it on its head, and make `osd_elements.c` the main unit, and create a `osd_utils.c` with utility functions that is included via a header file.